tools/dbTools.py

```diff
@@ -1,54 +1,177 @@
-import sqlite3
+import psycopg2
+import psycopg2.extras
 import logging
+import os
 
 
-class SQLite3Connector:
-    def __init__(self):
+class DatabaseConnector:
+    def __init__(self, database_url=None):
+        """Initialise an instance of the DatabaseConnector class and create:
+         - self.logger: a logger to log errors
+         - self.connection: the connection to the sqlite3 database
+         - self.cursor: the cursor to execute requests on the database
+        It also run the check_db method, creating the database if it doesn't
+        exists yet.
+        """
+
         self.logger = logging.getLogger(__name__)
-        self.connection = sqlite3.connect('db.sqlite3')
-        self.cursor = self.connection.cursor()
+        if not database_url:
+            database_url = os.getenv('DATABASE_URL')
+        self.connection = psycopg2.connect(database_url)
+        self.cursor = self.connection.cursor(
+            cursor_factory=psycopg2.extras.NamedTupleCursor
+        )
         self._check_db()
 
     def __del__(self):
+        """Commit all changes and close the database connection on the deletion
+        of this instance from memory.
+        """
+        self._close_all_spiders()
         self.connection.commit()
         self.cursor.close()
         self.connection.close()
 
+    def _close_all_spiders(self):
+        self._execute(
+            """
+            UPDATE spiders
+            SET status = %s, end_time = CURRENT_TIMESTAMP;
+            """,
+            ('finished',)
+        )
+
     def _execute(self, query, params=()):
+        """Try to execute the SQL query passed by the query parameter, with the
+        arguments passed by the tuple argument params.
+        """
         try:
             self.cursor.execute(query, params)
-        except sqlite3.Error as error:
+            self.connection.commit()
+        except psycopg2.Error as e:
             self.logger.error(
                 'An exception had been encountered when executing %s',
                 query,
             )
             raise
 
     def _check_db(self):
+        """Create the tables needed by the web scraper if they don't exists."""
         self._execute(
             """
             CREATE TABLE IF NOT EXISTS article
             (
-                id INTEGER PRIMARY KEY,
-                title VARCHAR(64),
+                id SERIAL PRIMARY KEY,
```

Comment:
this change sound specific to the database you use. in the future probably worth abstracting the database connectors for different databases. So having a PGSQLconnector and a SQLlite connect that share some functions but overwrite others. not important for now. worth adding to the backlog.